providers/flagsmith/src/main/java/dev.openfeature.contrib.providers.flagsmith/FlagsmithProvider.java

```
@@ -0,0 +1,263 @@
+            ? flagsmith.getEnvironmentFlags()
+            // Todo add traits when attributes are added to context
+            : flagsmith.getIdentityFlags(ctx.getTargetingKey());
+    }
+
+    /**
+     * Using the Flagsmith SDK this method resolves any type of flag into
+     * a ProviderEvaluation. Since Flagsmith's sdk is agnostic of type
+     * the flag needs to be cast to the correct type for OpenFeature's
+     * ProviderEvaluation object.
+     *
+     * @param key          the string identifier for the flag being resolved
+     * @param defaultValue the backup value if the flag can't be resolved
+     * @param ctx          an EvaluationContext object with flag evaluation options
+     * @param expectedType the expected data type of the flag as a class
+     * @param <T>          the data type of the flag
+     * @return a ProviderEvaluation object for the given flag type
+     * @throws OpenFeatureError when flag evaluation fails
+     */
+    private <T> ProviderEvaluation<T> resolveFlagsmithEvaluation(
+        String key, T defaultValue, EvaluationContext ctx, Class<?> expectedType
+    ) throws OpenFeatureError {
+        T flagValue = null;
+        ErrorCode errorCode = null;
+        Reason reason = null;
+        String variationType = "";
+        try {
+
+            Flags flags = getFlags(ctx);
+            // Check if the flag is enabled, return default value if not
+            boolean isFlagEnabled = flags.isFeatureEnabled(key);
+            if (!isFlagEnabled) {
+                return buildEvaluation(defaultValue, null, Reason.DISABLED, null);
+            }
+
+            Object value = flags.getFeatureValue(key);
+            // Convert the value received from Flagsmith.
+            flagValue = convertValue(value, expectedType);
+
+        } catch (FlagsmithClientError flagsmithApiError) {
+            return buildEvaluation(defaultValue, ErrorCode.GENERAL, Reason.ERROR, null);
+        }
```

You may want to consider simply throwing here instead of returning an evaluation. The SDK will handle setting the error and returning the default according to which `OpenFeatureError` impl you throw.

---

Switched it to throwing the exceptions now instead